ssl/quic/quic_rstream.c

```diff
@@ -135,3 +184,96 @@ int ossl_quic_rstream_available(QUIC_RSTREAM *qrs, size_t *avail, int *fin)
 #endif
     return 1;
 }
+
+int ossl_quic_rstream_get_record(QUIC_RSTREAM *qrs,
+                                 const unsigned char **record, size_t *rec_len,
+                                 int *fin)
+{
+    const unsigned char *record_ = NULL;
+    size_t rec_len_, max_len;
+
+    if (!ossl_sframe_list_lock_head(&qrs->fl, &qrs->head_range, &record_, fin)) {
+        *record = NULL;
+        *rec_len = 0;
+        return 0;
+    }
+
+    rec_len_ = (size_t)(qrs->head_range.end - qrs->head_range.start);
+
+    if (record_ == NULL && rec_len_ != 0) {
+        record_ = ring_buf_get_ptr(&qrs->rbuf, qrs->head_range.start,
+                                   &max_len);
+        if (!ossl_assert(record_ != NULL))
+            return 0;
+        if (max_len < rec_len_) {
+            rec_len_ = max_len;
+            qrs->head_range.end = qrs->head_range.start + max_len;
+        }
+    }
+
+    *rec_len = rec_len_;
```

Comment:
The "if" above suggests "rec_len_" could be 0 here - which means we return 1 on 0 length here, but 0 on 0 length above.

Reply:
Yep, I changed this to be consistent.